consensus/types/src/checkpoint.rs

```diff
@@ -29,6 +27,64 @@ pub struct Checkpoint {
     pub root: Hash256,
 }
 
+/// Use a custom implementation of SSZ to avoid the overhead of the derive macro.
```

Comment:
I can't spot the issue with this impl at a glance, but when I tried to sync a node with this change I got:

> Jan 20 06:23:41.955 WARN BlockProcessingFailure                  outcome: DBError(SszDecodeError(OffsetIntoFixedPortion(156))), msg: unexpected condition in processing block

I'm fairly sure it's this change, because I tried reverting the transaction changes, and the error persisted. And then when I reverted this change, it went away.

Reply:
Ah, it's because we're missing the impl of `ssz_fixed_len`:

https://docs.rs/ethereum_ssz/0.8.2/ssz/trait.Encode.html#method.ssz_fixed_len

Classic footgun, I think we've hit this multiple times. We should probably make that method mandatory, or combine it with `is_ssz_fixed_len` like:

```rust
enum SszLength {
   VariableLen,
   FixedLen(usize),
}

pub fn ssz_len() -> SszLength {
   SszLength::FixedLen(..)
}
```

Reply:
*Finally* catching up to this, thanks for the review work.

I went to implement `ssz_fixed_len` but I think the default implementation is right for us; `Transactions` is always variable length. I'll keep digging around and maybe figure our why I'm wrong 🤔

Reply:
Ooooh we're on `Checkpoint` not `Transaction`. You're right!